### PR TITLE
add support for keeping values as strings and add more options

### PR DESCRIPTION
**Pull request by programmerjake**
depends on https://github.com/kevinmehall/rust-vcd/pull/10 -- once that pull request is merged and a released version is made, I'll switch the Cargo.toml to it and change this pull request to non-draft.

### CODE REVIEW

**Comment by programmerjake** [comment]
progress is now blocked on https://github.com/kevinmehall/rust-vcd/issues/11 since https://github.com/kevinmehall/rust-vcd/pull/10 was merged.

File: src/map.rs [hunk 1]
```diff
@@ -43,32 +57,37 @@ impl Map {
         if self.paths.contains_key(&path) {
             panic!("oh no!")
         }
-        self.items.insert(item.code, vec![]);
+        let item = Rc::new(item);
+        self.items.insert(
+            item.code,
+            MapItemList {
+                values: vec![],
+                item: item.clone(),
```
Comment (sgpthomas) [inline src/map.rs]: Use `Rc::clone(&item)` instead of `.clone()` to make it clear that we are cloning the `Rc`

File: src/main.rs [hunk 2]
```diff
@@ -62,29 +142,40 @@ fn main() -> Result<(), Error> {
...
     }
 
     let mut output = Output::default();
 
-    for (path, var) in map.paths {
-        output.insert(path, &map.items[&var.code]);
+    let Map { mut items, paths } = map;
+
+    for (path, var) in paths {
+        output.insert(path, items.remove(&var.code).unwrap().values);
```
Comment (sgpthomas) [inline src/main.rs]: The remove here means that if `&var.code` shows up multiple times, the second instance will crash vcdump. Change this to `get()`

**Review by sgpthomas — CHANGES_REQUESTED** [review]
Thanks for doing this! Looks mostly good, except a small correctness issue and a style issue.

File: src/map.rs (hunk 1 above)
Comment (programmerjake) [reply src/map.rs]: done

File: src/main.rs (hunk 2 above)
Comment (programmerjake) [reply src/main.rs]: fixed

**Review by sgpthomas — APPROVED** [review]
LGTM!

**Pull request merged**